soda_machine.rb

```diff
@@ -7,12 +7,21 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.find { |soda| soda.brand == soda_brand }
   end
 
   def sell(soda_brand)
+    removed_soda, kept_sodas = @sodas.partition do |soda|
+      soda.brand == soda_brand
+    end
+    removed_soda.each do |soda|
+      @sodas.delete(soda)
+      @cash += soda.price
+    end
```

Comment:
You're throwing out all of the sodas that match?  If you have 10 pepsis, and I ask for a pepsi, you toss them all out of the machine, giving me one of them?

You've made this problem more complicated, and also given it different functionality (which the tests don't check for).